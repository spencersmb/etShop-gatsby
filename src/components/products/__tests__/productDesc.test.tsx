import ProductDescription from '@components/products/modules/productDesc'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	intro_title: 'product intro title',
	intro_description: 'intro desc'
}
describe('Product Description', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<ProductDescription {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct title', () => {
		const modalRender = render(<ProductDescription {...propsDefault}/>)
		expect(modalRender.getByTestId('title').innerHTML).toEqual(propsDefault.intro_title)
	})
	it('Should render correct description', () => {
		const modalRender = render(<ProductDescription {...propsDefault}/>)
		expect(modalRender.getByTestId('desc').innerHTML).toEqual(propsDefault.intro_description)
	})
})
