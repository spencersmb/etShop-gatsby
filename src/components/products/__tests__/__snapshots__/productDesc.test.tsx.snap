// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description renders correctly 1`] = `
Array [
  <span
    className="sc-bdVaJa jsByhr"
  >
    Description
  </span>,
  <h3
    className="sc-bwzfXH jrFMpq"
    data-testid="title"
  >
    product intro title
  </h3>,
  <div
    className="sc-htpNat fZCtkR"
    dangerouslySetInnerHTML={
      Object {
        "__html": "intro desc",
      }
    }
    data-testid="desc"
  />,
]
`;
